src/main/java/org/apache/commons/pool2/impl/GenericKeyedObjectPool.java

```diff
@@ -267,6 +272,25 @@ public GenericKeyedObjectPool(final KeyedPooledObjectFactory<K, T> factory,
         setConfig(config);
     }
 
+    /**
+     * Creates a new {@code GenericKeyedObjectPool} that tracks and destroys
+     * objects that are checked out, but never returned to the pool.
+     *
+     * @param factory   The object factory to be used to create object instances
+     *                  used by this pool
+     * @param config    The base pool configuration to use for this pool instance.
+     *                  The configuration is used by value. Subsequent changes to
+     *                  the configuration object will not be reflected in the
+     *                  pool.
+     * @param abandonedConfig  Configuration for abandoned object identification
+     *                         and removal.  The configuration is used by value.
+     */
```

Comment:
Just reviewing details: Please add `@since` tags to new public and protected methods.

Reply:
Added @since tags